website/ref/language.md

```diff
@@ -106,16 +106,75 @@ error.
 
 The three syntaxes above all evaluate to strings, and they are interchangeable.
 For instance, `xyz`, `'xyz'` and `"xyz"` are different syntaxes for the same
-string, and they are always equivalent.
+string, and they are always equivalent with the exception of
+**escape sequences** as documented above.
 
-Elvish does have a dedicated `float64` number type, but it does not have a
-literal syntax; instead it needs to be constructed using a function call, as in
-`put (float64 42)`; the argument `42` is a string. For convenience, all Elvish
-functions that expect number arguments also accept strings that can be parsed as
-a number, so explicit number constructors are not needed often.
+# Numeric data type
 
+Elvish has a numeric data type, `float64`, created using this syntax:
+`(float64 numeric-literal)`. For convenience, all Elvish functions that
+expect a numeric argument will accept a string that can be parsed as a
+number. They will automatically convert the string to a `float64`. Which
+means explicit `float64` constructors are almost never needed. Not even
+for special values like infinity (`Inf`) or not-a-number (`NaN`). Note
+that the letter case of these magic values doesn't matter so `inf` and
+`Inf` are equivalent.
 
-# List and Map
+Numeric literals can be specified using
+
+* decimal notation (e.g., `10`),
+* hexadecimal notation (e.g., `0xA`),
+* octal notation (e.g., `0o12`),
+* binary notation (e.g., `0b1010`),
+* floating point notation (e.g., `10.0`), and
+* scientific notation (e.g., `1.0e1`).
+
+All of those numeric literals are equivalent. However, note that all of
+those number literals are implicitly converted to the `float64` data type
+when used in a numeric context. The `float64` data type cannot represent
+many rational numbers exactly. So comparing numeric values that are derived
+from numerical operations rather than literals, as in the above examples,
+may not be strictly equal due to small rounding errors introduced by the
+numerical operations.
+
+To convert a `float64` data type to a string do `(to-string $float64_var)`
+or format it: `(printf '%.3f' $float64_var)`.
+
+# Exception data type
+
+Elvish has an exception data type, but it does not have a literal
+syntax for that type (despite the literal representation shown below by
+`put`). Instead it needs to be constructed using a function call, as in
+`(fail 'was unexpected')`. If an external command returns a non-zero exit
```

Comment:
This example should be `?(fail ...)`.

Reply:
No, because it shows how to throw/raise an exception; not its representation when shown using `put`. However, I agree the capture parens aren't appropriate in this context so I'll remove them.